pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageIdImpl.java

```diff
@@ -18,17 +18,11 @@
  */
 package org.apache.pulsar.client.impl;
 
-import com.google.common.collect.ComparisonChain;
 import org.apache.pulsar.client.api.MessageId;
 
 /**
  */
 public class BatchMessageIdImpl extends MessageIdImpl {
-    private final static int NO_BATCH = -1;
-    private final int batchIndex;
```

Comment:
Should we distinguish between a single Message and BatchMessage?
`batchIndex` and `batchSize` should be attributes of BatchMessage.

Reply:
Not sure I completely understand. So you think we should have another interface for `BatchMessageId`?

The core issue we're solving here is that it's possible to have both `MessageIdImpl` and `BatchMessageIdImpl` instances that refer to a single unbatched message. By moving the fields to `MessageIdImpl`, we can implement equals/hashCode in `MessageIdImpl` and have it work for the subclass `BatchMessageIdImpl`. There are other ways of achieving that I suppose, but since it wasn't public API I just did it that way because it seemed simpler.